### PR TITLE
Add Ruby 2.3.0 & 2.4.1 testing, remove Ruby 1.9.3 & 2.0 testing

### PR DESCRIPTION
**Pull request by thomasriley**
## Pull Request Checklist

sensu-plugins/sensu-plugins-feature-requests#27

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] RuboCop passes

- [x] Existing tests pass 

#### Purpose
- Add Ruby 2.3.0 & 2.4.1 testing
- Remove redundant Ruby 1.9.3 & 2.0 testing

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,8 +26,8 @@ deploy:
   on:
     tags: true
     all_branches: true
-    rvm: 1.9.3
-    rvm: 2.0
```
Comment (majormoses) [inline .travis.yml:30]: I am ok with this as it is an EOL version, I would just like to know what failed that made you want to remove this.

**Review by majormoses — APPROVED** [review]
LGTM

File: .travis.yml (hunk 1 above)
Comment (majormoses) [reply .travis.yml:30]: I see it was already removed in gemspec: https://github.com/sensu-plugins/sensu-plugins-mailgun/commit/e14928cad331bdf3e01acd9a9822954ac86388c3

Comment (thomasriley) [reply .travis.yml:30]: Nothing failed, however the required Ruby version in the gemspec for this plugin has already been bumped to 2.1.0 previously so I thought there was little point testing against 2.0.

Comment (thomasriley) [reply .travis.yml:30]: You beat me replying :)

**Comment by majormoses** [comment]
This plugin was also lacking love, I tried to tag old releases not previously tagged and added changelog messages (probably not accurate enough but did not want to spend too much time given that we should never have this again).

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-mailgun/versions/1.0.0

**Pull request merged**